src/app/reporting/LoggingManagement.cpp

```diff
@@ -0,0 +1,979 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2015-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "LoggingManagement.h"
+#include "LoggingConfiguration.h"
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/EventDataElement.h>
+#include <core/CHIPEventLoggingConfig.h>
+#include <core/CHIPTLVUtilities.hpp>
+#include <inttypes.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemTimer.h>
+
+using namespace chip::TLV;
+
+namespace chip {
+namespace Platform {
+// TODO: Implement CriticalSectionEnter per platform
```

Comment:
Could you actually implement?
These should be placed in a central place in chip/core